### PR TITLE
Implement api calls logger

### PR DESCRIPTION
**Pull request by Szeliga**
Adds an AuditedApi proxy object, that logs all API methods called into a an audit.log file

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -47,3 +47,7 @@ end
 group :development, :test do
   gem 'letter_opener'
 end
+
+group :test do
+  gem 'timecop'
```
Comment (lubieniebieski) [inline Gemfile]: let's add https://github.com/alindeman/zonebie as well

File: app/services/audited_api.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+class AuditedApi
+  attr_accessor :object, :logger
+
+  def initialize(object, logger = AuditLogger.new(Rails.root.join('log', 'audit.log')))
+    self.object = object
+    self.logger = logger
```
Comment (nekath) [inline app/services/audited_api.rb]: Do you reasign object or logger after initializing this class?
If not please narrow api - in this case use only readers.

Comment (Szeliga) [inline app/services/audited_api.rb]: Good point.

File: Gemfile (hunk 1 above)
Comment (Szeliga) [inline Gemfile]: Added

File: app/services/audited_api.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  def respond_to_missing?(method, *)
+    @object.respond_to?(method)
+  end
+
+  private
+
+  def log(method_name, response, arguments)
+    args = map_arguments(method_name, arguments)
+    result = SUCCESSFUL_STATUS_CODES.include?(status_code(response).to_i) ? 'OK' : 'ERROR'
+    "#{username} -- #{result} -- #{I18n.t(method_name, args)}"
```
Comment (lubieniebieski) [inline app/services/audited_api.rb]: how can we be sure that there's always a translation for a given method?

**Comment by lubieniebieski** [comment]
:blue_heart:

**Pull request merged**